src/EventStore.Client/EventStoreClientConnectivitySettings.cs

```diff
@@ -74,11 +82,25 @@ public class EventStoreClientConnectivitySettings {
 		/// True if pointing to a single EventStoreDB node.
 		/// </summary>
 		public bool IsSingleNode => GossipSeeds.Length == 0;
-
+		
 		/// <summary>
 		/// True if communicating over an insecure channel; otherwise false.
 		/// </summary>
-		public bool Insecure { get; set; }
+		public bool Insecure {
+			get {
+				return IsSingleNode
+					? string.Equals(Address.Scheme, Uri.UriSchemeHttp, StringComparison.OrdinalIgnoreCase)
+					: _insecure;
+			}
+			set {
+				_insecure = value;
+
+				_defaultAddress = new UriBuilder {
```

Comment:
Changing `Insecure` should not result in changing the set address

Reply:
It's only setting the default address which should be returned incase there is no address set. It won't change the address once it's set. This is done to keep the `Address.Scheme` in sync with the `Insecure` setting.

Reply:
Refactored it a bit to resemble this behavior.